pkg/transform/oauth/oauth.go

```diff
@@ -143,7 +145,7 @@ func Translate(identityProviders []IdentityProvider) (*CRD, []secrets.Secret, er
 		}
 		oauthCrd.Spec.IdentityProviders = append(oauthCrd.Spec.IdentityProviders, idP)
 
-		if secret.Metadata.Name != httpdAuthSecret || p.Kind == httpProviderKind {
+		if (secret.Metadata.Name != httpdAuthSecret || p.Kind == httpProviderKind) && p.Kind != basicAuthProviderKind {
```

Comment:
I might have missed something but I would use the above switch entries to handle those cases.

Reply:
Yes. this needs some attention, I think. I added the the line above with the intention of coming back to figure out how to better handle this.